### PR TITLE
Fix assertion error on gemm_splitk_benchmark.py

### PR DESCRIPTION
**Pull request by LiyangLingIntel**
The cause of this error is that we should initialize the results tensor when using `atomic_add`, otherwise it would read dirty memory from previous benchmarking cases.

### CODE REVIEW

File: benchmarks/triton_kernels_benchmark/gemm_splitk_benchmark.py [hunk 1]
```diff
@@ -128,6 +128,7 @@ def forward(ctx, a, b, c, acc_dtype=None):
             [512, 32768, 8192],
             [1024, 28672, 8192],
             [3072, 4096, 3072],
+            [4096, 4096, 4096],
```
Comment (anmyachev) [inline benchmarks/triton_kernels_benchmark/gemm_splitk_benchmark.py:131]: Adding a new combination breaks the CI and seems a bit out of topic for this pull request. Maybe we should move this change to a separate pull request?

Comment (etiotto) [reply benchmarks/triton_kernels_benchmark/gemm_splitk_benchmark.py:131]: Agree. Split that change in a separate PR please @LiyangLingIntel

Comment (etiotto) [reply benchmarks/triton_kernels_benchmark/gemm_splitk_benchmark.py:131]: Other than that LGTM.

**Review by etiotto — APPROVED** [review]

File: benchmarks/triton_kernels_benchmark/gemm_splitk_benchmark.py (hunk 1 above)
Comment (whitneywhtsang) [reply benchmarks/triton_kernels_benchmark/gemm_splitk_benchmark.py:131]: I disagree, as this PR supposes to fix the 4k functional error.

**Review by whitneywhtsang — COMMENTED** [review]
please add `gemm_splitk_shape_4096_4096_4096` to XeTLA to fix the CI failure.

File: benchmarks/triton_kernels_benchmark/gemm_splitk_benchmark.py (hunk 1 above)
Comment (LiyangLingIntel) [reply benchmarks/triton_kernels_benchmark/gemm_splitk_benchmark.py:131]: I've added 4k shape to XeTLA splitk list, benchmark CI works now.

**Review by chengjunlu — APPROVED** [review]

**Pull request merged**